buildpacktoml_test.go

```diff
@@ -0,0 +1,745 @@
+				} else if !strings.Contains(err.Error(), "some error") {
+					t.Fatalf("Incorrect error: %s\n", err)
+				}
+			})
+
+			it("should error when the command fails", func() {
+				env.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+				if err := os.RemoveAll(platformDir); err != nil {
+					t.Fatalf("Error: %s\n", err)
+				}
+				_, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+				if err, ok := err.(*lifecycle.Error); !ok || err.Type != lifecycle.ErrTypeBuildpack {
+					t.Fatalf("Incorrect error: %s\n", err)
+				}
+			})
+
+			when("modifying the env fails", func() {
+				var appendErr error
+
+				it.Before(func() {
+					appendErr = errors.New("some error")
+				})
+
+				each(it, []func(){
+					func() {
+						env.EXPECT().AddRootDir(gomock.Any()).Return(appendErr)
+					},
+					func() {
+						env.EXPECT().AddRootDir(gomock.Any()).Return(nil)
+						env.EXPECT().AddRootDir(gomock.Any()).Return(appendErr)
+					},
+					func() {
+						env.EXPECT().AddRootDir(gomock.Any()).Return(nil)
+						env.EXPECT().AddRootDir(gomock.Any()).Return(nil)
+						env.EXPECT().AddEnvDir(gomock.Any()).Return(appendErr)
+					},
+					func() {
+						env.EXPECT().AddRootDir(gomock.Any()).Return(nil)
+						env.EXPECT().AddRootDir(gomock.Any()).Return(nil)
+						env.EXPECT().AddEnvDir(gomock.Any()).Return(nil)
+						env.EXPECT().AddEnvDir(gomock.Any()).Return(appendErr)
+					},
+				}, "should error", func() {
+					env.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+					mkdir(t,
+						filepath.Join(appDir, "layers-A-v1", "layer1"),
+						filepath.Join(appDir, "layers-A-v1", "layer2"),
+					)
+					mkfile(t, "build = true",
+						filepath.Join(appDir, "layers-A-v1", "layer1.toml"),
+						filepath.Join(appDir, "layers-A-v1", "layer2.toml"),
+					)
+					if _, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config); err != appendErr {
+						t.Fatalf("Incorrect error: %s\n", err)
+					}
+				})
+			})
+
+			it("should error when launch.toml is not writable", func() {
+				env.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+				mkdir(t, filepath.Join(layersDir, "A", "launch.toml"))
+				if _, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config); err == nil {
+					t.Fatal("Expected error")
+				}
+			})
+
+			it("should error when the launch bom has a top level version", func() {
+				env.EXPECT().WithPlatform(platformDir).Return(append(os.Environ(), "TEST_ENV=Av1"), nil)
+				mkfile(t,
+					"[[bom]]\n"+
+						`name = "some-dep"`+"\n"+
+						`version = "some-version"`+"\n",
+					filepath.Join(appDir, "launch-A-v1.toml"),
+				)
+				_, err := bpTOML.Build(lifecycle.BuildpackPlan{}, config)
+				h.AssertNotNil(t, err)
+				expected := "top level version which is deprecated"
+				h.AssertStringContains(t, err.Error(), expected)
+			})
+
+			it("should error when the build bom has a top level version", func() {
```

Comment:
It might make more sense to fail out during export, but it's very convenient to test here :/

Reply:
Might as well fail early and save the user some time, especially if it is convenient :)